### ISSUE TITLE
Training result performance issue

### ISSUE THREAD

**Issue by popov1212** [Rex-sys-hk/PlanScope#6]
I use 34 epoch in trainning, but the result video shows some problem.

the video of 12 epoch result performed better than the video of 18 epoch result. Have you ever met this problem? The 12epoch result can turn right smoothly at the cross scenario but then running out of lane bound
The 18epoch result can hardly turn right at the cross scenario. It's actually stopped at the cross scenario.

I only change two parameters in sim_scope.sh
BUILDER=nuplan_mini
FILTER=mini_demo_scenario
That is to say I only use this scenario to test my training result

**Comment by popov1212** [comment]
And I also run the results you opened: PlanScope-h10-m6 in the same scenario:
BUILDER=nuplan_mini
FILTER=mini_demo_scenario
The result vedio shows: after ego car turns right, It suddenly stoped in the straight lane while the gt smothly go forward.

Why the ego car stopped suddenly in the straight lane while expert keeps running in const velocity?
I wonder if there is something wrong with my parameter setting?

**Comment by popov1212** [comment]
Thanks for your help

**Comment by popov1212** [comment]
Thanks for your reply 

I wonder if the nuplan have the close loop train? 

And how could you usually do to treat the absense of target velocity in the dataset?

By the way what is the mean of "baseline" in your last reply?

Thanks again for your kindly answer

**Comment by Rex-sys-hk** [comment]
1. It's different framework. Some work might adopt ppo RL in closed loop simulation to finetune the performance.
2. It's an open question. Maybe you can try to address it. But it is meaning less in realworld adoptation 
3. Baseline means other methods in comparison.

**Comment by popov1212** [comment]
Thank you for your reply
 could you recommend some articles about ppo RL？

**Comment by Rex-sys-hk** [comment]
In our experiments, for 20% dataset it is 35 epoch, for 100% dataset it is 25 epoch.
h10 is the detail horizon, m6 is mode number. You can check our paper for more details.

**Comment by popov1212** [comment]
thanks for your kindly reply

Could you recommend some relevant open source about ppo RL？

**Comment by Rex-sys-hk** [comment]
Maybe you can start from here https://hrl.boyuai.com/chapter/2/ppo%E7%AE%97%E6%B3%95.

**Comment by popov1212** [comment]
Thanks for your kindly reply


The trainining process showed total loss decreasing, but minADE not decreasing. I think that is the key problem of causing the failure of performance.

Please checkout the attached picture.

Thanks in advance for your answer!

**Comment by popov1212** [comment]
![IMG_1592](https://github.com/user-attachments/assets/9111394f-7daf-4e8e-afd5-14bc4f7c89cf)
![IMG_1591](https://github.com/user-attachments/assets/af57b0b3-bd55-4196-b3ee-7748bdb1827d)

**Comment by popov1212** [comment]
![IMG_1591](https://github.com/user-attachments/assets/737fa095-29e7-4224-bc14-4d1c6dd96176)
![IMG_1592](https://github.com/user-attachments/assets/237b2ec3-91d0-427f-9824-48d65f1c5729)

**Comment by popov1212** [comment]
Besides, I use the full dataset, that is to say 1 millio n scenarios are used to train.

**Comment by Rex-sys-hk** [comment]
Your training script looks good. But you can try setting
> model.cat_x=false model.ref_free_traj=false.
 which might help your training process more stable.

Besides, if you decide to train on full dataset, you need
> data_loader.datamodule.train_fraction=1.0
> data_loader.datamodule.val_fraction=1.0
> data_loader.datamodule.test_fraction=1.0

**Comment by popov1212** [comment]
Thanks 

other yaml need to be change?

Like default_training.yaml? custom_lightning?

**Comment by Rex-sys-hk** [comment]
You do not need to. Changes in train.sh will change configurations throughout.

**Comment by popov1212** [comment]
Thanks for your answer

I wonder how to modified the weights of minADE minFDE PredAvgADE PredAvgFDE MR?

I could not find where is the definition of above variable in PlanScope code

**Comment by popov1212** [comment]
I have setting parameters as you told me

model.cat_x=false model.ref_free_traj=false
data_loader.datamodule.train_fraction=1.0
data_loader.datamodule.val_fraction=1.0
data_loader.datamodule.test_fraction=1.0

for the full data training,but seems didn't work.

But I didn't run the 
"=== Start Sanity Ckeck==="

Before 
"===Start Training==="

Is Sanity Ckeck matters?

**Comment by Rex-sys-hk** [comment]
Are you using the cached dataset?

**Comment by Rex-sys-hk** [comment]
> "=== Start Sanity Ckeck==="

does not matter.

**Comment by Rex-sys-hk** [comment]
Kindly advise you to use mode_num 1 and 20% dataset to debug.

**Comment by popov1212** [comment]
> Are you using the cached dataset?

Yes! 

I put the full dataset(train_boston train_pittsburgh train_vegas_1-train_vegas_6 val ) under the path: nuplan/dataset/nuplan-v1.1/trainval
Then run the feature cache script 

Below is the feature cache script I run
export PYTHONPATH=$PYTHONPATH:$(pwd)

python run_training.py
py_func=cache +training=train_scope
scenario_builder=nuplan
cache.cache_path=/home/my_name/nuplan/exp/cache_pluto_1M
cache.cleanup_cache=false
scenario_filter=training_scenarios_1M
worker.threads_per_node=40

Finally, I run the train script, and the content is same with above I have attached 2 days ago

**Comment by popov1212** [comment]
> Kindly advise you to use mode_num 1 and 20% dataset to debug.

Do you mean setting the parameter
num_modes: 1 in scope_model.yaml
And set 
data_loader.datamodule.train_fraction=0.2
data_loader.datamodule.val_fraction=0.2
data_loader.datamodule.test_fraction=0.2
In train script.

Am I right?

**Comment by popov1212** [comment]
> > Are you using the cached dataset?
> 
> Yes!
> 
> I put the full dataset(train_boston train_pittsburgh train_vegas_1-train_vegas_6 val ) under the path: nuplan/dataset/nuplan-v1.1/trainval Then run the feature cache script
> 
> Below is the feature cache script I run export PYTHONPATH=$PYTHONPATH:$(pwd)
> 
> python run_training.py py_func=cache +training=train_scope scenario_builder=nuplan cache.cache_path=/home/my_name/nuplan/exp/cache_pluto_1M cache.cleanup_cache=false scenario_filter=training_scenarios_1M worker.threads_per_node=40
> 
> Finally, I run the train script, and the content is same with above I have attached 2 days ago

Anything wrong with my steps above?

**Comment by Rex-sys-hk** [comment]
All the settings seems right. We are also setting up our experiments again on new devices and trying to reproduce and address above problems.

**Comment by popov1212** [comment]
Many thanks for your support 
This problem troublemed me for long.

**Comment by popov1212** [comment]
And plsease try the script I post above.

Maybe easier to reproduce the problem

**Comment by Rex-sys-hk** [comment]
If it is still not solved, please have me your email address and I will share the preprocessed dataset with you. This might be a cause of up mentioned problems.

**Comment by Rex-sys-hk** [comment]
We still failed to reproduce your problems after thorough tests. One potential cause is the NATTEN version. We advise to use NATTEN≈0.14.6.
